### PR TITLE
subchart version

### PR DESCRIPTION
**Pull request by mikolajmdnv**

### CODE REVIEW

File: charts/kubeit-app-of-apps/templates/applications.yaml [hunk 1]
```diff
@@ -99,15 +109,13 @@ spec:
...
-        - ./appValues/{{ $app.name }}/environment/{{ $envDirectory }}/region/{{ $.Values.region }}/cluster/{{ $.Values.clusterColour }}.yaml
-        - ./appValues//{{ $app.name }}/environment/{{ $envDirectory }}/region/{{ $.Values.region }}/cluster/namespaces/{{ . }}.yaml
+        - "{{ $.Values.globals.valuesPath }}/global.yaml"
+        {{- range $valuesList }}
+        - "{{ . }}"
+        {{- end }}
       values: |
         app: {{ $app.name }}
+        namespace: {{ . }}
         defaultRouting:
```
Comment (mikolajmdnv) [inline charts/kubeit-app-of-apps/templates/applications.yaml:119]: Add merging defaultRouting map with `$allValues`

File: charts/kubeit-app-of-apps/templates/applications.yaml [hunk 2]
```diff
@@ -17,10 +17,10 @@
...
-{{- $globalScope := printf "appValues/%s/appGlobal.yaml" $app.name }}
-{{- $envScope := printf "appValues/%s/environment/%s/%s.yaml" $app.name $.Values.env $.Values.env }}
-{{- $regionScope := printf "appValues/%s/environment/%s/region/%s/%s.yaml" $app.name $.Values.env $.Values.region $.Values.region }}
-{{- $clusterScope := printf "appValues/%s/environment/%s/region/%s/cluster/%s.yaml" $app.name $.Values.env $.Values.region $.Values.clusterColour }}
+{{- $globalScope := printf "%s/appValues/%s/appGlobal.yaml" $.Values.globals.valuesPath $app.name }}
+{{- $envScope := printf "%s/appValues/%s/environment/%s/%s.yaml" $.Values.globals.valuesPath $app.name $envDirectory $envDirectory }}
+{{- $regionScope := printf "%s/appValues/%s/environment/%s/region/%s/%s.yaml" $.Values.globals.valuesPath $app.name $envDirectory $.Values.region $.Values.region }}
+{{- $clusterScope := printf "%s/appValues/%s/environment/%s/region/%s/cluster/%s.yaml" $.Values.globals.valuesPath $app.name $envDirectory $.Values.region $.Values.clusterColour }}
 {{- $valuesList := list $globalScope $envScope $regionScope $clusterScope }}
 # ArgoCD Image Updater Variables
```
Comment (mikolajmdnv) [inline charts/kubeit-app-of-apps/templates/applications.yaml:25]: Check ArgoCD Image Updater vars

File: charts/kubeit-app-of-apps/templates/applications.yaml [hunk 3]
```diff
@@ -99,15 +109,13 @@ spec:
...
-        - ./appValues/{{ $app.name }}/appGlobal.yaml
-        - ./appValues/{{ $app.name }}/environment/{{ $envDirectory }}/{{ $envDirectory }}.yaml
-        - ./appValues/{{ $app.name }}/environment/{{ $envDirectory }}/region/{{ $.Values.region }}/{{ $.Values.region }}.yaml
-        - ./appValues/{{ $app.name }}/environment/{{ $envDirectory }}/region/{{ $.Values.region }}/cluster/{{ $.Values.clusterColour }}.yaml
-        - ./appValues//{{ $app.name }}/environment/{{ $envDirectory }}/region/{{ $.Values.region }}/cluster/namespaces/{{ . }}.yaml
+        - "{{ $.Values.globals.valuesPath }}/global.yaml"
+        {{- range $valuesList }}
+        - "{{ . }}"
+        {{- end }}
       values: |
```
Comment (mikolajmdnv) [inline charts/kubeit-app-of-apps/templates/applications.yaml:116]: Add a variable which would contain a list with nonempty valuesFiles to see which ones were actually loaded

File: charts/kubeit-app-of-apps/templates/applications.yaml [hunk 4]
```diff
@@ -99,15 +109,13 @@ spec:
...
         app: {{ $app.name }}
```
Comment (mikolajmdnv) [inline charts/kubeit-app-of-apps/templates/applications.yaml:117]: Maybe move passed vars from bootstrap to global variables

File: charts/kubeit-app-of-apps/templates/applications.yaml [hunk 5]
```diff
@@ -59,6 +59,15 @@
 {{- end }}
 {{- end }}
```
Comment (mikolajmdnv) [inline charts/kubeit-app-of-apps/templates/applications.yaml:60]: Move all rendred vars/templates to `_helpers.tpl`

File: charts/kubeit-app-of-apps/templates/applications.yaml [hunk 6]
```diff
@@ -59,6 +59,15 @@
...
 
+# Create a map with Helm Values inline
+{{ $allValues := dict }}
+{{- range $filename := $valuesList -}}
+  {{ $allValues = mergeOverwrite $allValues ($.Files.Get . | fromYaml) }}
```
Comment (mikolajmdnv) [inline charts/kubeit-app-of-apps/templates/applications.yaml:65]: Check if valuesFiles are accessiable when app of apps chart is referenced as subchart

File: charts/kubeit-app-of-apps/templates/applications.yaml (hunk 2 above)
Comment (nikodemjedynak-dnv) [reply charts/kubeit-app-of-apps/templates/applications.yaml:25]: Works as expected, tested locally on all scopes (appGlobal, env, region, cluster) - it is defined only per application

**Pull request closed without merging**